crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -1919,9 +1934,25 @@ async fn init_sim_executor(
         None,
     );
 
+    // Take the last 4 digits of the db name and use it as the port number
+    let base_port = db_name
+        .chars()
+        .rev()
+        .take(4)
+        .collect::<String>()
+        .chars()
+        .rev()
+        .collect::<String>();
```

Comment:
With a fairly small probability, your hashing approach might produce a number that's smaller than 4 digits, at which point this will fail.  The approach of accepting the full hash number would fix this.